plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/MigrateProcedure.java

```diff
@@ -315,8 +321,22 @@ private Schema toIcebergSchema(List<Column> columns)
         return new Schema(icebergSchema.asStructType().fields());
     }
 
-    private static org.apache.iceberg.types.Type toIcebergType(Type type, AtomicInteger nextFieldId)
+    private static org.apache.iceberg.types.Type toIcebergType(Type type, AtomicInteger nextFieldId, HiveStorageFormat storageFormat)
     {
+        // TODO https://github.com/trinodb/trino/issues/20481
+        if (type instanceof TimestampType && storageFormat == AVRO) {
+            throw new TrinoException(NOT_SUPPORTED, "Migrating timestamp type with %s format is not supported.".formatted(storageFormat));
+        }
+
+        if (type.equals(TIMESTAMP_MILLIS)) {
+            if (storageFormat == PARQUET) {
+                return Types.TimestampType.withZone();
+            }
+            if (storageFormat == ORC) {
+                return Types.TimestampType.withoutZone();
+            }
+        }
```

Comment:
```
if (type instanceof TimestampType) {
            if (storageFormat == PARQUET) {
                return Types.TimestampType.withZone();
            }
            if (storageFormat == ORC) {
                return Types.TimestampType.withoutZone();
            }
            throw new TrinoException(NOT_SUPPORTED, "Migrating timestamp type with %s format is not supported.".formatted(storageFormat));
}
```
I think we want to support migrating any precision timestamp, not just millis

Reply:
Could you check why `testMigrateTimestampMicrosHiveTableWithParquet` and `testMigrateTimestampNanosHiveTableWithParquet` are not failing with your current code ?